File: backend/modules/mission.py
from backend.modules import mining, discovery, event_engine


def status():
    workers_payload = mining.workers()
    workers = workers_payload.get("workers", [])

    total_hashrate = sum(float(w.get("hashrate") or 0) for w in workers)
    online_workers = len([w for w in workers if float(w.get("hashrate") or 0) > 0])

    scan = discovery.scan()
    systems = scan.get("systems", [])

    events = event_engine.live()[-50:]
    warnings = len([e for e in events if e.get("severity") == "warning"])
    critical = len([e for e in events if e.get("severity") == "critical"])

    return {
        "overall": "healthy" if critical == 0 else "critical",
        "miners": {
            "online": online_workers,
            "total": len(workers),
            "hashrate": total_hashrate
        },
        "infrastructure": {
            "systems": len(systems),
            "healthy": len(systems)
        },
        "services": {
            "api": "online",
            "miningcore": workers_payload.get("status", "unknown"),
            "stratum": "active" if online_workers > 0 else "idle",
            "discovery": "online"
        },
        "events": {
            "warnings": warnings,
            "critical": critical
        }
    }

File: frontend/js/mission.js
function missionFormatHashrate(value) {
  value = Number(value || 0);
  if (value >= 1e12) return `${(value / 1e12).toFixed(2)} TH/s`;
  if (value >= 1e9) return `${(value / 1e9).toFixed(2)} GH/s`;
  return `${value.toFixed(0)} H/s`;
}

function ensureMissionBar() {
  if (document.getElementById("missionBar")) return;

  const bar = document.createElement("section");
  bar.id = "missionBar";
  bar.className = "mission-bar";
  bar.innerHTML = "Loading mission status...";

  const nav = document.getElementById("topNav");
  if (nav && nav.parentNode) nav.insertAdjacentElement("afterend", bar);
  else document.body.prepend(bar);
}

async function loadMissionStatus() {
  ensureMissionBar();

  try {
    const res = await fetch("/api/mission/status");
    if (!res.ok) return;

    const s = await res.json();
    const critical = s.events?.critical || 0;
    const warnings = s.events?.warnings || 0;

    const bar = document.getElementById("missionBar");
    bar.classList.toggle("critical", critical > 0);
    bar.classList.toggle("warning", critical === 0 && warnings > 0);

    bar.innerHTML = `
      <a href="/analytics.html">🟢 ${missionFormatHashrate(s.miners?.hashrate)} Fleet</a>
      <a href="/graph.html">🟢 ${s.miners?.online}/${s.miners?.total} ASICs Online</a>
      <a href="/pools.html">🟢 MiningCore ${String(s.services?.miningcore || "unknown").toUpperCase()}</a>
      <a href="/discovery.html">🟢 ${s.infrastructure?.systems} Systems</a>
      <a href="/analytics.html">🟢 Stratum ${String(s.services?.stratum || "unknown").toUpperCase()}</a>
      <a href="/analytics.html">⚠ ${warnings} Warnings</a>
      <a href="/analytics.html">🚨 ${critical} Critical</a>
    `;
  } catch {}
}

window.addEventListener("DOMContentLoaded", () => {
  loadMissionStatus();
  setInterval(loadMissionStatus, 3000);
});
